### PR TITLE
Fix BrowserHttpClientAdapter canceled hangs

### PR DESCRIPTION
**Pull request by morlay**
### New Pull Request Checklist

- [ ] I have read the [Documentation](https://pub.dartlang.org/packages/dio)
- [ ] I have searched for a similar pull request in the [project](https://github.com/flutterchina/dio/pulls) and found none
- [ ] I have updated this branch with the latest `develop` to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I am adding
- [ ] I have updated the documentation (if necessary)
- [ ] I have run the tests and they pass

This merge request fixes / refers to the following issues: ...

### Pull Request Description

could check case https://github.com/morlay/roundtripper/runs/4862938300?check_suite_focus=true

When using `BrowserHttpClientAdapter` cancel hangs until timeout. `DefaultHttpClientAdapter` work well.
```
00:39 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:40 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:41 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:42 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:43 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:44 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:45 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:46 +4: test/basic_test.dart: cancel                                                                                                                                                                 
00:46 +4 -1: test/basic_test.dart: cancel [E]                                                                                                                                                          
  TimeoutException after 0:00:30.000000: Test timed out after 30 seconds. See https://pub.dev/packages/test#timeouts
  package:test_api                                 Invoker._handleError.<fn>
  org-dartlang-sdk:///lib/async/zone.dart 1420:46  StaticClosure._rootRun
  org-dartlang-sdk:///lib/async/zone.dart 1327:34  _CustomZone.run
  package:test_api                                 Invoker.heartbeat.<fn>.<fn>
  org-dartlang-sdk:///lib/async/zone.dart 1428:12  StaticClosure._rootRun
  org-dartlang-sdk:///lib/async/zone.dart 1327:34  _CustomZone.run
```

When patch the code in this PR, everything fine.

### CODE REVIEW

**Review by kuhnroyal — APPROVED** [review]
Can you add a test for this?
The change seems to make sense but I have not used cancels on web yet.

**Comment by morlay** [comment]
@kuhnroyal 

It could be run with `dart test --platform=chrome` for web.

not need to add more testing.

may just consider to enable it in ci

like https://github.com/dart-lang/http/blob/master/.github/workflows/test-package.yml#L63

**Comment by morlay** [comment]
@kuhnroyal 
I upgrade test option `testOn` to disable some tests for `vm only`
then CI could be run `vm` & `chrome` both

File: dio/test/basic_test.dart [hunk 1]
```diff
@@ -46,7 +45,7 @@ void main() {
       Dio().get('http://http.invalid'),
       throwsA((e) => e is DioError && e.error is SocketException),
     );
-  });
+  }, testOn: "vm");
```
Comment (morlay) [inline dio/test/basic_test.dart:48]: May fix next PR， to normalize the error between Default & Browser HttpAdapter

Comment (kuhnroyal) [reply dio/test/basic_test.dart:48]: That would be a great improvement for 5.0

**Comment by kuhnroyal** [comment]
Thank you, nice work!

**Pull request merged**